discovery-infra/test_infra/tools/static_ips.py

```diff
@@ -12,50 +12,82 @@ def generate_macs(count):
 
 
 def generate_static_ips_data_from_tf(tf_folder):
+
     tfvars_json_file = os.path.join(tf_folder, consts.TFVARS_JSON_NAME)
+
     with open(tfvars_json_file) as _file:
         tfvars = json.load(_file)
-    return _generate_static_ips_data(tfvars['machine_cidr_addresses'][0],
-                                     tfvars['provisioning_cidr_addresses'][0],
+
+    return _generate_static_ips_data(tfvars['machine_cidr_addresses'],
+                                     tfvars['provisioning_cidr_addresses'],
                                      tfvars['libvirt_master_macs'],
                                      tfvars['libvirt_secondary_master_macs'],
                                      tfvars['libvirt_worker_macs'],
                                      tfvars['libvirt_secondary_worker_macs'])
 
 
-def _generate_static_ips_data(machine_cidr,
-                              provisioning_cidr,
+def _generate_static_ips_data(machine_cidrs,
+                              provisioning_cidrs,
                               masters_macs,
                               masters_secondary_macs,
                               workers_macs,
                               workers_secondary_macs):
+
     num_masters = len(masters_macs)
-    num_workers = len(workers_macs)
 
-    # set starting static ips
-    masters_static_starting_ip = str(ipaddress.ip_address(ipaddress.IPv4Network(machine_cidr).network_address) + 30)
-    masters_static_secondary_starting_ip = str(ipaddress.ip_address(ipaddress.IPv4Network(provisioning_cidr).network_address) + 30)
-    workers_static_starting_ip = str(ipaddress.ip_address(ipaddress.IPv4Network(machine_cidr).network_address) + 30 + num_masters)
-    workers_static_secondary_starting_ip = str(ipaddress.ip_address(ipaddress.IPv4Network(provisioning_cidr).network_address) + 30 + num_masters)
+    static_ips = []
+
+    static_ips.extend(_genetate_ips(masters_macs, machine_cidrs, 0))
+
+    static_ips.extend(_genetate_ips(
+        masters_secondary_macs, provisioning_cidrs, 0))
+
+    static_ips.extend(_genetate_ips(workers_macs, machine_cidrs, num_masters))
+
+    static_ips.extend(_genetate_ips(workers_secondary_macs,
+                                    provisioning_cidrs, num_masters))
 
-    # set static ips lists
-    masters_static_ips = utils.create_ip_address_list(num_masters, masters_static_starting_ip)
-    masters_static_secondary_ips = utils.create_ip_address_list(num_masters, masters_static_secondary_starting_ip)
-    workers_static_ips = utils.create_ip_address_list(num_workers, workers_static_starting_ip)
-    workers_static_secondary_ips = utils.create_ip_address_list(num_workers, workers_static_secondary_starting_ip)
+    return static_ips
+
+
+def _genetate_ips(mac_addresses, cidrs, address_offset=0):
+
+    num_nodes = len(mac_addresses)
+
+    first_net = ip_network(cidrs[0])
+    ipv4_gen = _static_conf_gen(
```

Comment:
The new line is coming in a bad place. It is difficult to understand what the "if" is referring to.  I gave an example of how to use it in the next comment.